### PR TITLE
asyncly: add package

### PR DESCRIPTION
**Pull request by gjasny**
Specify library name and version:  **asyncly/cci.20230420**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by gjasny** [comment]
Blocked by #17472

File: recipes/asyncly/all/test_v1_package/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+cmake_minimum_required(VERSION 3.15)
```
Comment (uilianries) [inline recipes/asyncly/all/test_v1_package/CMakeLists.txt:1]: The test_v1_package is no longer mandatory, you can remove if you want: https://github.com/conan-io/conan-center-index/discussions/17406

**Review by uilianries — COMMENTED** [review]
Looking good, only few comments.

File: recipes/asyncly/all/test_v1_package/CMakeLists.txt (hunk 1 above)
Comment (gjasny) [reply recipes/asyncly/all/test_v1_package/CMakeLists.txt]: removed

File: recipes/asyncly/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("boost/1.81.0", transitive_headers=True)
+        self.requires("function2/4.2.2", transitive_headers=True)
+        self.requires("prometheus-cpp/1.1.0", transitive_headers=True)
```
[the remark this answers is not in the document]
Comment (uilianries) [reply recipes/asyncly/all/conanfile.py:57]: @gjasny Thank you for showing the error, indeed the project is using a dependency as public header ... not good but happens ...

**Review by uilianries — APPROVED** [review]
LGTM

**Review by SSE4 — APPROVED** [review]

**Pull request merged**